### DISCUSSION TITLE
Error: error: error unmarshaling JSON: while decoding JSON: json: unknown field "jobs"

### DISCUSSION THREAD

**Discussion by shailendra118** [concourse/concourse#7100] [Help & Support]
Hi Team,

I am trying to build a pipeline in Concourse and codebase in private GitLab, when I try to execute the below command in Fly and getting the below error. Also, even tried to downloaded the code (https://github.com/bijukunjummen/ci-concourse-caching-sample)  and executed it, got the same below error.

Please help.

$ fly -t gcp execute -c pipeline.yml
error: error unmarshaling JSON: while decoding JSON: json: unknown field "jobs"

**Comment by taylorsilva** [comment]
`fly execute` is for running a task and it looks like you're trying to pass in an entire pipeline. A task is part of a pipeline and a pipeline can contain one or many tasks, which are wrapped up in [`jobs`](https://concourse-ci.org/jobs.html).

A task config has all the fields on this page: https://concourse-ci.org/tasks.html
There's even an example on how to use `fly execute` to run a task: https://concourse-ci.org/tasks.html#running-tasks

Hope that helps!

**Reply by taylorsilva** [reply]
This might also help you see the difference between a pipeline and task config yaml: https://concourse-ci.org/common-pipeline-practices.html#putting-task-configs-in-files
In that example, you can `fly execute` the `todays-date.yml`. Trying to `fly execute` the `pipeline.yml` will return the error you're currently getting.
